samtranslator/model/api/http_api_generator.py

```diff
@@ -308,11 +308,16 @@ def _construct_basepath_mappings(self, basepaths, http_api):
                 invalid_regex = r"[^0-9a-zA-Z\/\-\_]+"
                 if re.search(invalid_regex, path) is not None:
                     raise InvalidResourceException(self.logical_id, "Invalid Basepath name provided.")
-                # ignore leading and trailing `/` in the path name
-                m = re.search(r"[a-zA-Z0-9]+[\-\_]?[a-zA-Z0-9]+", path)
-                path = m.string[m.start(0) : m.end(0)]
-                if path is None:
-                    raise InvalidResourceException(self.logical_id, "Invalid Basepath name provided.")
+
+                if path == "/":
+                    path = ""
+                else:
+                    # ignore leading and trailing `/` in the path name
+                    m = re.search(r"[a-zA-Z0-9]+[\-\_]?[a-zA-Z0-9]+", path)
+                    path = m.string[m.start(0) : m.end(0)]
```

Comment:
It's not clear from reading the code what `path` is, and why it's reassigned to in such a way (e.g. `foo-bar-baz` becomes `foo-bar`... _why_?).

If it's generating a logical ID from an arbitrary string, we should refactor in such a way; it seems to be a fairly generic use-case.

Reply:
That's a good point. I am not sure about the reason behind creating logical_id like this. But we can try to understand where we use the logical id further and modify the logic accordingly.

I will merge this PR for now and can track this later.